youtube_dl/extractor/medaltv.py

```diff
@@ -0,0 +1,127 @@
+        webpage = self._download_webpage(url, video_id)
+
+        hydration_data = self._search_regex(r'<script>.*hydrationData\s*=\s*({.+?})\s*</script>', webpage, 'hydration data', default='{}')
+        parsed = self._parse_json(hydration_data, video_id)
+
+        author_info = next(iter(parsed.get('profiles').values()))
+        clip_info = parsed.get('clips').get(video_id)
+
+        if('error' in clip_info):
+            error_code = clip_info.get('error')
+            if(error_code == 404):
+                raise ExtractorError('That clip does not exist.', expected=True, video_id=video_id)
+            else:
+                raise ExtractorError('An unknown error occurred ({}).'.format(error_code), video_id=video_id)
+
+        source_aspect_ratio = clip_info.get('sourceWidth') / clip_info.get('sourceHeight')
+
+        # ordered from worst to best quality
+        heights = (144, 240, 360, 480, 720, 1080)
+
+        formats = []
+        thumbnails = []
+
+        for height in heights:
+            format_key = '{}p'.format(height)
+            video_key = 'contentUrl{}'.format(format_key)
+            thumbnail_key = 'thumbnail{}'.format(format_key)
+            width = int(round(source_aspect_ratio * height))
+
+            # second condition needed as sometimes medal says
+            # they have a format when in fact it is another format
+            if(video_key in clip_info and format_key in clip_info.get(video_key)):
+                formats.append({
+                    'url': clip_info.get(video_key),
+                    'format_id': format_key,
+                    'width': width,
+                    'height': height
+                })
+
+            if(thumbnail_key in clip_info and format_key in clip_info.get(thumbnail_key)):
+                thumbnails.append({
+                    'id': format_key,
+                    'url': clip_info.get(thumbnail_key),
+                    'width': width,
+                    'height': height
+                })
+
+        # add source to formats
+        formats.append({
+            'url': clip_info.get('contentUrl'),
+            'format_id': 'source',
+            'width': clip_info.get('sourceWidth'),
+            'height': clip_info.get('sourceHeight')
+        })
+
+        return {
+            'id': video_id,
+            'title': clip_info.get('contentTitle'),
+
+            'formats': formats,
+            'url': clip_info.get('contentUrl'),
+
+            'thumbnails': thumbnails,
+            'thumbnail': try_get(thumbnails, lambda x: x[-1]['url'], compat_str),
+
+            'description': clip_info.get('contentDescription'),
+
+            # author information
+            'uploader': author_info.get('displayName'),
+            'timestamp': float_or_none(clip_info.get('created') / 1000),
+            'uploader_id': author_info.get('id'),
+            'uploader_url': 'https://medal.tv/users/{}'.format(author_info.get('id')),
```

Comment:
will return `https://medal.tv/users/None` if an `author id` couldn't be extracted.

Reply:
Resolved by checking if author id is present:
```python
author_url = 'https://medal.tv/users/{}'.format(author_info.get('id')) if author_info.get('id') else None
```